recipes/install.rb

```diff
@@ -100,8 +101,19 @@
       else
         "http://packages.treasuredata.com/#{major}/redhat/$releasever/$basearch"
       end
+    when '3'
+      if platform == "amazon"
+        amazon_version = node['platform_version'] == '2' ? '2' : '1'
+        if node['td_agent']['yum']['amazon'][amazon_version].include? node['platform_version']
+          package_version = node['platform_version']
+        else
+          package_version = node['td_agent']['yum']['amazon'][amazon_version].last
```

Comment:
Would this mean that the cookbook installs the latest built packages if the platform is something unknown....? I don't think it's appropriate behavior.

Even if you want the cookbook to behave so, it should be opt-in explicitly by declaring custom attribute, e.g. `node["td_agent"]["yum_amazon_releasever"]`. I think some change like follows would be sufficient to use dedicated repository for Amazon Linux. You'd need to declare `node["td_agent"]["yum_amazon_releasever"]` by your own if you'd like to install td-agent3 for unsupported Amazon Linux versions, though.

```rb
when '3'
  if platform == "amazon
    amazon_version = node["platform_version"]
    "http://packages.treasuredata.com/#{major}/amazon/#{amazon_version}/#{node["td_agent"]["yum_amazon_releasever"]}/$basearch"
  else
    "http://packages.treasuredata.com/#{major}/redhat/$releasever/$basearch"
  end
  ( ... )
```

cc: @repeatedly